x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternActionTests.java

```diff
@@ -171,4 +172,51 @@ public void testInnerPut_existingLeaderIndicesAndAutoFollowMetadata() {
         assertThat(autoFollowMetadata.getHeaders().get("name1"), notNullValue());
     }
 
+    public void testInnerPutWithExcludingPatterns() {
```

Comment:
Would it be an option to do this in `testInnerPut` and `testInnerPut_existingLeaderIndices`? Seems like those do not validate that the leader exclusion patterns are not set afterwards now, which seems wrong.